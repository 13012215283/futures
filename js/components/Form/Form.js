import React from 'react';
import { StyleSheet, View } from 'react-native';
import PropTypes from 'prop-types';
import { px2dp } from '../../utils/px2dp';

export default function Form({ children }) {
  return (
    <View style={{ flex: 1 }}>
      <View style={styles.container}>
        {React.Children.map(children, child => child)}
      </View>
    </View>
  );
}
Form.propTypes = {
  children: PropTypes.node.isRequired,
};

const styles = StyleSheet.create({
  container: {
    margin: px2dp(64),
    marginBottom: 0,
  },
});
