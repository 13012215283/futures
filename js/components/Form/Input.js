import React from 'react';
import PropTypes from 'prop-types';

import { Text, TextInput, View, StyleSheet } from 'react-native';
import { px2dp } from '../../utils/px2dp';
import noop from '../../utils/noop';

import { colors, sizes } from '../../components/themes';

const Input = ({
  label,
  placeholder,
  val,
  children,
  onChangeText,
  ...props
}) => (
  <View style={styles.containerStyle}>
    <View style={[{ justifyContent: 'center' }, styles.labelWrapperStyle]}>
      <Text style={[styles.labelText, { justifyContent: 'space-between' }]}>
        {label}
      </Text>
    </View>
    <View style={[{ flex: 1, justifyContent: 'center' }]}>
      <TextInput
        style={{ fontSize: sizes.f2 }}
        underlineColorAndroid="transparent"
        placeholder={placeholder}
        placeholderTextColor={colors[1102]}
        onChangeText={onChangeText}
        {...props}
      />
    </View>
    {React.Children.map(children, child => (
      <View style={styles.center}>{child}</View>
    ))}
  </View>
);

Input.propTypes = {
  label: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  val: PropTypes.string,
  children: PropTypes.node,
  onChangeText: PropTypes.func,
};

Input.defaultProps = {
  children: null,
  val: '',
  onChangeText: noop,
};

export default Input;

const styles = StyleSheet.create({
  center: {
    alignItems: 'flex-end',
    justifyContent: 'center',
  },
  containerStyle: {
    flexDirection: 'row',
    borderBottomWidth: 1,
    borderBottomColor: colors[1104],
    height: px2dp(104),
  },
  labelWrapperStyle: {
    width: px2dp(88),
  },
  labelText: {
    color: colors[1101],
    fontSize: sizes.f2,
    textAlign: 'justify',
  },
});
