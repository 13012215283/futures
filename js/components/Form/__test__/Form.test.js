import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Form from '../Form';

describe('<Form/>', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Form>
          <Text>hello</Text>
        </Form>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders multiple children correctly', () => {
    const tree = shallow(
      <Form>
        <Text>HELLO</Text>
        <Text>WORLD</Text>
        <Text>!!!!!</Text>
      </Form>
    );
    expect(tree.find(Text).length).toBe(3);
  });
});
