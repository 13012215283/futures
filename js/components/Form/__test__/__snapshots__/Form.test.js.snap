// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form/> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "margin": 64,
        "marginBottom": 0,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      hello
    </Text>
  </View>
</View>
`;
